radar-commons-android/src/main/java/org/radarbase/android/util/StorageLevelReceiver.kt

```diff
@@ -0,0 +1,41 @@
+package org.radarbase.android.util
+
+/**
+ * Keep track of storage levels.
+ */
+class StorageLevelReceiver(private val listener: ((StorageState) -> Unit)? = null ) {
+
+    /** Latest storage level in [0.75,1]
+     *  where 0.75 indicates partial, 75% of storage is full
+     *  and 1 indicates complete storage is full
+     */
+    private var level: Float = 0.0f
+        private set
+
+    var state: StorageState = StorageState.AVAILABLE
+        private set
+
+    fun setLevel(level: Float) {
+        this.level = level
+        setState()
+    }
+
+    private fun setState() {
```

Comment:
Instead of setting, its just updating the state based on the level
```suggestion
    private fun updateState() {
```